### PR TITLE
Improve error reporting in BlockMachineProcessor

### PR DESCRIPTION
**Pull request by georgwiese**
Extracted from #2275.

This PR makes it easier to debug failing code generation in `BlockMachineProcessor` by printing the code generated so far.

Example:
```
$ RUST_LOG=debug cargo run pil test_data/std/binary_large_test.asm -o output -f
...
Code generation failed for connection:
  main::instr_and $ [0, main::X0, main::X1, main::X2] is main_binary::latch * main_binary::sel[0] $ [main_binary::operation_id, main_binary::A, main_binary::B, main_binary::C]
Known arguments:
  main_binary::operation_id
  main_binary::A
  main_binary::B
Error:
  Unable to derive algorithm to compute output value "main_binary::C"
The following code was generated so far:
main_binary::sel[0][3] = 1;
main_binary::operation_id[3] = params[0];
main_binary::A[3] = params[1];
main_binary::B[3] = params[2];
main_binary::operation_id[2] = main_binary::operation_id[3];
main_binary::operation_id_next[2] = main_binary::operation_id[3];
main_binary::operation_id[1] = main_binary::operation_id[2];
main_binary::operation_id_next[1] = main_binary::operation_id[2];
main_binary::operation_id[0] = main_binary::operation_id[1];
main_binary::operation_id_next[0] = main_binary::operation_id[1];
...
```

### CODE REVIEW

File: executor/src/witgen/jit/test_util.rs [hunk 1]
```diff
@@ -1,42 +1 @@
-use itertools::Itertools;
```
Comment (georgwiese) [inline executor/src/witgen/jit/test_util.rs:1]: Yes, this module is empty now. But #2276 adds to it, so if that gets merge first, it won't be empty on `main`.

Comment (leonardoalt) [reply executor/src/witgen/jit/test_util.rs:1]: weird

**Review by leonardoalt — DISMISSED** [review]

**Comment by leonardoalt** [comment]
Approved, but might need conflict resolving when the other PR is merged.

File: executor/src/witgen/data_structures/mutable_state.rs [hunk 2]
```diff
@@ -46,18 +46,14 @@ impl<'a, T: FieldElement, Q: QueryCallback<T>> MutableState<'a, T, Q> {
...
-            Ok(m) => m,
-            // Calling unwrap() would require KnownMachine to implement Debug.
-            Err(e) => {
-                panic!("Expected exactly one machine with substring '{substring}', but found {e}.",)
-            }
-        }
+            .map_err(|e| {
+                format!("Expected exactly one machine with substring '{substring}', but found {e}.")
+            })
+            .unwrap()
```
Comment (georgwiese) [inline executor/src/witgen/data_structures/mutable_state.rs:56]: Got this suggestion in another PR: https://github.com/powdr-labs/powdr/pull/2276#discussion_r1899495511

**Comment by georgwiese** [comment]
Ready for review again.

**Review by leonardoalt — APPROVED** [review]

**Pull request merged**